File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLongestSubstring = function(s) {
    let st = new Set();
    let right=0;
    let left=0;
    let ans=0;
    while(right<s.length){
        if(st.has(s[right])){
            st.delete(s[left]);
            left++;
        }else{
            st.add(s[right]);
            ans = Math.max(ans,right-left+1);
            right++;
        }
    }
    return ans;
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
     unordered_set<char>st;
     int l=0,r=0,ans=0;
     while(r<s.size()){
        if(st.find(s[r]) != st.end()){
            st.erase(s[l]);
            l++;
        }else{
            st.insert(s[r]);
            ans = max(ans,r-l+1);
            r++;
        }
     }
     return ans;
    }
};